sdk/keyvault/azcertificates/client.go

```diff
@@ -84,82 +82,11 @@ func (b BeginCreateCertificateOptions) toGenerated() *generated.KeyVaultClientCr
 
 // CreateCertificateResponse contains response fields for Client.BeginCreateCertificate
 type CreateCertificateResponse struct {
-	Operation
-}
-
-// CreateCertificatePoller is the poller returned by the Client.BeginCreateCertificate
-type CreateCertificatePoller struct {
-	certName       string
-	certVersion    string
-	vaultURL       string
-	client         *generated.KeyVaultClient
-	createResponse CreateCertificateResponse
-	lastResponse   generated.KeyVaultClientGetCertificateResponse
-	getRawResponse *http.Response
-	resumeToken    string
-}
-
-// ResumeToken returns a token for resuming polling at a later time
-func (p *CreateCertificatePoller) ResumeToken() (string, error) {
-	return p.resumeToken, nil
-}
-
-// Done returns true if the LRO has reached a terminal state
-func (b *CreateCertificatePoller) Done() bool {
-	if b.getRawResponse == nil {
-		return false
-	}
-	return b.getRawResponse.StatusCode == http.StatusOK
-}
-
-// Poll fetches the latest state of the operations. It returns an HTTP response or error.
-// If the LRO has completed successfully, the poller's state is updated and the HTTP response is returned.
-// If the LRO has completed with failure or was cancelled, the poller's state is updated and the error is returned.)
-func (b *CreateCertificatePoller) Poll(ctx context.Context) (*http.Response, error) {
-	var getRawResp *http.Response
-	ctx = runtime.WithCaptureResponse(ctx, &getRawResp)
-	resp, err := b.client.GetCertificate(ctx, b.vaultURL, b.certName, b.certVersion, nil)
-	if err == nil {
-		b.getRawResponse = getRawResp
-		b.lastResponse = resp
-		b.createResponse.ID = b.lastResponse.ID
-		return getRawResp, nil
-	}
-
-	if getRawResp != nil && getRawResp.StatusCode == http.StatusNotFound {
-		// The certificate has not been fully created yet
-		b.getRawResponse = getRawResp
-		b.lastResponse = resp
-		return b.getRawResponse, nil
-	}
-
-	// There was an error in this operation, return the original raw response and the error
-	return getRawResp, err
-}
-
-// FinalResponse returns the final response after the operations has finished
-func (b *CreateCertificatePoller) FinalResponse(ctx context.Context) (CreateCertificateResponse, error) {
-	return b.createResponse, nil
-}
-
-// PollUntilDone continually polls the service with a 't' delay until completion.
-func (b *CreateCertificatePoller) PollUntilDone(ctx context.Context, t time.Duration) (CreateCertificateResponse, error) {
-	for {
-		resp, err := b.Poll(ctx)
-		if err != nil {
-			return CreateCertificateResponse{}, err
-		}
-		b.getRawResponse = resp
-		if b.Done() {
-			break
-		}
-		time.Sleep(t)
-	}
-	return b.createResponse, nil
+	CertificateWithPolicy
```

Comment:
Note that we now return the created certificate with the LRO completes instead of this intermediate `Operation` type.